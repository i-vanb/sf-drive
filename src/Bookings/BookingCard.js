import star from "../img/star.png";
import calendarIcon from "../img/calendarIcon.png";
import {toPriceView} from "../utils/toPriceView";
import React from "react";


export const BookingCard = ({mark, model, year, price, begin, end}) => {
    return(
        <div className="user-cars__item">
            <div className="user-cars__item__photo">
                <img src="/cars/img/imgCar1.png"/>
            </div>
            <div className="user-cars__item__desc">
                <div className="collage_car__main__reviews" style={{marginBottom: "10px"}}>
                    <img src={star}/>
                    <span>4,5</span>
                    <span>(4)</span>
                </div>
                <div><img src={calendarIcon} /></div>
                <div>{mark} {model}, {year}</div>
                <div>{toPriceView(price)} руб</div>
            </div>
        </div>
    )
}
