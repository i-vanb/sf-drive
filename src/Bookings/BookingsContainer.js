import React from "react";
import {connect} from "react-redux";


class BookingsContainer extends React.Component {
    render() {
        return <div>Booking<Test /></div>
    }
}

const mapStateToProps = state => ({
    state: state.auth
})

export default connect(mapStateToProps, {})(BookingsContainer)



const Test = () => {
    testfunc()

    return<div></div>
}
const testfunc = async () => {
    const promise = new Promise(((resolve, reject) => {
        resolve('returning')
    }))
    console.log(promise)
    // const array1 = [1,2,3].map(async (e, i) => {
    //     new Promise((resolve => {
    //         setTimeout(()=>resolve, 5000)
    //     }))
    //     return `response${e}`
    // })
    // console.log(array1)
}
