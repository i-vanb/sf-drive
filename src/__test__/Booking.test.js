import {shallow} from "enzyme";
import {BookingCard} from "../Bookings/BookingCard";

const mockProps = {mark:'', model:'', year:'', price:'', begin:'', end:''}
const wrapper = shallow(<BookingCard {...mockProps} />);

describe('Booking', () => {
    it('should be defined', () => {
        expect(wrapper).toBeDefined()
    });
    it('snapshot matching', () => {
        expect(wrapper).toMatchSnapshot()
    });
});
