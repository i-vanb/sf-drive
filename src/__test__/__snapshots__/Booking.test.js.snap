// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking snapshot matching 1`] = `
<div
  className="user-cars__item"
>
  <div
    className="user-cars__item__photo"
  >
    <img
      src="/cars/img/imgCar1.png"
    />
  </div>
  <div
    className="user-cars__item__desc"
  >
    <div
      className="collage_car__main__reviews"
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      <img
        src="star.png"
      />
      <span>
        4,5
      </span>
      <span>
        (4)
      </span>
    </div>
    <div>
      <img
        src="calendarIcon.png"
      />
    </div>
    <div>
       
      , 
    </div>
    <div>
       руб
    </div>
  </div>
</div>
`;
